cloud/scope/object_storage_bucket.go

```diff
@@ -101,15 +101,46 @@ func (s *ObjectStorageBucketScope) AddFinalizer(ctx context.Context) error {
 
 // ApplyAccessKeySecret applies a Secret containing keys created for accessing the bucket.
 func (s *ObjectStorageBucketScope) ApplyAccessKeySecret(ctx context.Context, keys [NumAccessKeys]linodego.ObjectStorageKey, secretName string) error {
+	var secretStringData map[string]string
+
+	// If the secret is of ClusterResourceSet type, encapsulate real data in the outer data
+	if s.Bucket.Spec.SecretType == "addons.cluster.x-k8s.io/resource-set" {
+		bucketDataSecret := `kind: Secret
```

Comment:
instead of doing this templating, could we create a normal `secret` object and just convert that to a string for the access-keys-secret value?